r2/errors.go

```diff
@@ -60,3 +60,11 @@ type NotFoundError struct{ msg string }
 func NewNotFoundError(msg string) NotFoundError { return NotFoundError{msg: msg} }
 
 func (e NotFoundError) Error() string { return e.msg }
+
+// AuthError represents an error due to missing or invalid auth information.
+type AuthError struct{ msg string }
```

Comment:
This doesn't need to be exported no? Can also do `type authError string`.

Reply:
Good point. Can do that for all of the errors actually.

Reply:
Re 1: is exporting the constructor not sufficient if you just need to initialize it outside the package?